test/integration/client-side-encryption/client_side_encryption.range.prose.ts

```diff
@@ -0,0 +1,534 @@
+      precision: 2
+    }),
+    factory: value => new Decimal128(value.toString())
+  },
+  {
+    type: 'DoubleNoPrecision',
+    rangeOptions: prepareOptions({ sparsity: { $numberLong: '1' } }),
+    factory: value => new Double(value)
+  },
+  {
+    type: 'DoublePrecision',
+    rangeOptions: prepareOptions({
+      min: { $numberDouble: '0' },
+      max: { $numberDouble: '200' },
+      sparsity: { $numberLong: '1' },
+      precision: 2
+    }),
+    factory: value => new Double(value)
+  },
+  {
+    type: 'Date',
+    rangeOptions: prepareOptions({
+      min: { $date: { $numberLong: '0' } },
+      max: { $date: { $numberLong: '200' } },
+      sparsity: { $numberLong: '1' }
+    }),
+    factory: value => new Date(value)
+  },
+  {
+    type: 'Int',
+    rangeOptions: prepareOptions({
+      min: { $numberInt: '0' },
+      max: { $numberInt: '200' },
+      sparsity: { $numberLong: '1' }
+    }),
+    factory: value => value
+  },
+  {
+    type: 'Long',
+    rangeOptions: prepareOptions({
+      min: { $numberLong: '0' },
+      max: { $numberLong: '200' },
+      sparsity: { $numberLong: '1' }
+    }),
+    factory: value => Long.fromNumber(value)
+  }
+];
+
+const basePath = '/test/spec/client-side-encryption/etc/data';
+
+const readEncryptedFieldsFile = (dataType: string): Promise<string> =>
+  readFile(join(__dirname, '../../..', basePath, `range-encryptedFields-${dataType}.json`), {
+    encoding: 'utf8'
+  });
+
+describe('Range Explicit Encryption', function () {
+  let clientEncryption;
+  let keyId;
+  let keyVaultClient;
+  let encryptedClient;
+  let encryptedZero;
+  let encryptedSix;
+  let encryptedThirty;
+  let encryptedTwoHundred;
+  let compareNumericValues;
+  for (const { type: dataType, rangeOptions, factory } of dataTypes) {
+    context(`datatype ${dataType}`, async function () {
+      beforeEach(async function () {
+        compareNumericValues = function (value: unknown, expected: number): void {
+          if (dataType === 'DoubleNoPrecision' || dataType === 'DoublePrecision') {
+            expect(value).to.equal(expected);
+          } else if (dataType === 'Long') {
+            expect(value).to.equal(expected);
+          } else {
+            expect(value).to.deep.equal(factory(expected));
+          }
+        };
+        const ClientEncryption = this.configuration.mongodbClientEncryption.ClientEncryption;
+        const keyDocument1 = EJSON.parse(
+          await readFile(join(__dirname, '../../../', basePath, 'keys', `key1-document.json`), {
+            encoding: 'utf8'
+          }),
+          {
+            relaxed: false
+          }
+        ) as unknown as Document;
+
+        // confirm that this is an ObjectId
+        keyId = keyDocument1._id;
+        const encryptedFields = EJSON.parse(await readEncryptedFieldsFile(dataType), {
+          relaxed: false
+        }) as unknown as Document;
+
+        const utilClient: MongoClient = await this.configuration.newClient().connect();
```

Comment:
missed one connect usage here

Reply:
intentional - util client that's closed in the before (not a part of tests).  do you still want me to remove it?